### ISSUE TITLE
Odditities when going from two to one screen (with b33fb2a0)

### ISSUE THREAD

**Issue by juw** [awesomeWM/awesome#1002]
Hi,

on Debian Sid with b33fb2a0 (and also earlier versions) from current master (with at least LuaJIT) I have the following problem:

1. I have a dual screen setup:
  ```
Screen 0: minimum 8 x 8, current 3286 x 1080, maximum 32767 x 32767
LVDS1 connected primary 1366x768+0+0 (normal left inverted right x axis y axis) 280mm x 160mm
   1366x768      60.02*+
   1360x768      59.80    59.96  
   1280x720      60.00  
   1024x768      60.00  
   1024x576      60.00  
   960x540       60.00  
   800x600       60.32    56.25  
   864x486       60.00  
   640x480       59.94  
   720x405       60.00  
   680x384       60.00  
   640x360       60.00  
DP1 disconnected (normal left inverted right x axis y axis)
DP2 disconnected (normal left inverted right x axis y axis)
DP3 disconnected (normal left inverted right x axis y axis)
HDMI1 disconnected (normal left inverted right x axis y axis)
HDMI2 disconnected (normal left inverted right x axis y axis)
HDMI3 disconnected 1920x1080+1366+0 (normal left inverted right x axis y axis) 0mm x 0mm
VGA1 disconnected (normal left inverted right x axis y axis)
VIRTUAL1 disconnected (normal left inverted right x axis y axis)
  1920x1080 (0x4e) 148.500MHz +HSync +VSync
        h: width  1920 start 2008 end 2052 total 2200 skew    0 clock  67.50KHz
        v: height 1080 start 1084 end 1089 total 1125           clock  60.00Hz
  ```
  Everything looks fine, even when the second screen is not attached but I did not update the screen configuration with xrandr just yet. (See [screenshot](http://rbw.goe.net/jw/2screens.png)).

2. If I now switch of the second screen:
  ```
Screen 0: minimum 8 x 8, current 1366 x 768, maximum 32767 x 32767
LVDS1 connected primary 1366x768+0+0 (normal left inverted right x axis y axis) 280mm x 160mm
   1366x768      60.02*+
   1360x768      59.80    59.96  
   1280x720      60.00  
   1024x768      60.00  
   1024x576      60.00  
   960x540       60.00  
   800x600       60.32    56.25  
   864x486       60.00  
   640x480       59.94  
   720x405       60.00  
   680x384       60.00  
   640x360       60.00  
DP1 disconnected (normal left inverted right x axis y axis)
DP2 disconnected (normal left inverted right x axis y axis)
DP3 disconnected (normal left inverted right x axis y axis)
HDMI1 disconnected (normal left inverted right x axis y axis)
HDMI2 disconnected (normal left inverted right x axis y axis)
HDMI3 disconnected (normal left inverted right x axis y axis)
VGA1 disconnected (normal left inverted right x axis y axis)
VIRTUAL1 disconnected (normal left inverted right x axis y axis)
  ```
  the status icons disappear (though the space where they were is still occupied) and the right click menu in firefox does not respect the bounds of the window anymore (see [screenshot](http://rbw.goe.net/jw/1screen.png)).

Cheers.

**Comment by psychon** [comment]
No idea why the tray icons disappear.

For the right-click menu... I don't know, but I would guess that this is a bug in Firefox. Awesome isn't involved in where it places its menu. Does it work correctly after a firefox restart?

**Comment by juw** [comment]
No, it does not work after a firefox restart. It only works correctly
again, if I either restart X or add the second monitor again.

**Comment by psychon** [comment]
How about restarting awesome? Also, your `xrandr` output confuses me. There is only a single "connected" entry, everything else is "disconnected" and yet you have two screens?!?

**Comment by juw** [comment]
Sorry, I should have been more clear: xrandr does not show the second
screen, because it was disconnected at that point, but for all intents
and purposes (softwarewise) it is still present. So VIRTUAL1 is the
second screen.

Restarting awesome brings no change to the firefox situation but it
brings back the icons. 

Am Sat, 16 Jul 2016 03:00:57 -0700
schrieb Uli Schlachter <notifications@github.com>:

**Comment by psychon** [comment]
Placement of right-click-menus seems to be here in GTK: https://git.gnome.org/browse/gtk+/tree/gtk/gtkmenu.c?id=e9078e3abb90aa107f3c329940fa1e7454a31fcb#n4339
This does something based on the monitor's workarea...
The function for getting the workarea doesn't seem to be doing too much: https://git.gnome.org/browse/gtk+/tree/gdk/x11/gdkmonitor-x11.c?id=e9078e3abb90aa107f3c329940fa1e7454a31fcb#n61
https://git.gnome.org/browse/gtk+/tree/gdk/x11/gdkscreen-x11.c?id=e9078e3abb90aa107f3c329940fa1e7454a31fcb#n236

Just by starting at the code, I don't see anything here. My only suggestion would be "could you check if the same problem occurs when running an X11 session without any window manager?" (mostly because this would mean that the bug is in GTK, but I don't really know where in GTK it would be)

**Comment by funeralismatic** [comment]
Bug is in GTK+ 3.x. Not only with Firefox and not only on (re)attach/remove screens (but only in awesome?)

GTK+ 3.x is a bug by itself.

**Comment by juw** [comment]
Sorry for the silence, I had no access to a second monitor this week.

Actually, my firefox still uses GTK+2 but the code also looks, like it
is not the problem.

I was able to reproduce it also with awesome 3.5.9 and then checked
with a plain X11 session: The problem does not occur in a X11 session
without any WM, since the right-click menu does not draw there at all.

Surprisingly right now it works and I am not able to reproduce it
anymore.

**Comment by juw** [comment]
Scratch that last sentence, I am able to reproduce it again.
